core/src/main/java/com/sequenceiq/cloudbreak/conclusion/step/NetworkCheckerConclusionStep.java

```diff
@@ -0,0 +1,71 @@
+import com.sequenceiq.cloudbreak.client.RPCResponse;
+import com.sequenceiq.cloudbreak.node.status.NodeStatusService;
+
+@Component
+public class NetworkCheckerConclusionStep extends ConclusionStep {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworkCheckerConclusionStep.class);
+
+    @Inject
+    private NodeStatusService nodeStatusService;
+
+    @Override
+    public Conclusion check(Long resourceId) {
+        RPCResponse<NodeStatusReport> networkReport = nodeStatusService.getNetworkReport(resourceId);
+        if (networkReport.getResult() == null) {
+            String nodeStatusResult = networkReport.getFirstTextMessage();
+            LOGGER.info("Network report result was null, original message: {}", nodeStatusResult);
+            return succeeded();
+        }
+
+        List<String> networkFailures = new ArrayList<>();
+        List<String> networkFailureDetails = new ArrayList<>();
+        checkNetworkFailures(networkReport, networkFailures, networkFailureDetails);
+
+        if (networkFailures.isEmpty()) {
+            return succeeded();
+        } else {
+            return failed(networkFailures.toString(), networkFailureDetails.toString());
+        }
+    }
+
+    private void checkNetworkFailures(RPCResponse<NodeStatusReport> networkReport, List<String> networkFailures, List<String> networkFailureDetails) {
+        for (NodeStatus nodeStatus : networkReport.getResult().getNodesList()) {
+            NetworkDetails networkDetails = nodeStatus.getNetworkDetails();
+            String host = nodeStatus.getStatusDetails().getHost();
+            if (networkDetails.getCcmEnabled() && !HealthStatus.OK.equals(networkDetails.getCcmAccessible())) {
+                networkFailures.add("CCM is not accessible from node " + host + ". Please check network settings!");
+                String details = String.format("CCM health status is %s for node %s", networkDetails.getCcmAccessible(), host);
+                networkFailureDetails.add(details);
+                LOGGER.warn(details);
+            }
+            if (!HealthStatus.OK.equals(networkDetails.getClouderaComAccessible())) {
+                networkFailures.add("Cloudera.com is not accessible from node: " + host + ". Please check network settings!");
+                String details = String.format("Cloudera.com accessibility status is %s for node %s", networkDetails.getClouderaComAccessible(), host);
+                networkFailureDetails.add(details);
+                LOGGER.warn(details);
+            }
+            if (!HealthStatus.OK.equals(networkDetails.getAnyNeighboursAccessible())) {
+                networkFailures.add("Node " + host + " cannot reach any neighbour nodes. Please check nodes and network settings!");
+                String details = String.format("Neighbours accessibility status is %s for node %s", networkDetails.getAnyNeighboursAccessible(), host);
```

Comment:
the background network checks (for nodestatus monitor) neighbours are not checked at all, it's costly (as happens from every host) +  some tools can consider that as an  attack (?) (+ if the network is not set up correctly that could ping nodes that are not inside even part of the cluster, that happened with GCP)

Reply:
Thanks for clarification! What is your suggestion? Skip this neighbour checking or maybe use a different endpoint for that?

Reply:
the response object won't change here ... maybe the way should be which is used to ping other hosts (from every hosts)...at the moment this operation is a network scanning, if we could change that to do the scanning against a list of the exact host (if it stored somehow on the nodes??) or if salt can do similar checks between minions it can be even in a different endpoint